### PR TITLE
HIVE-12698 : introduce HiveAuthorizationTranslator interface for isolating authori…

### PR DESCRIPTION
**Pull request by thejasmn**
…zation impls from hive internal classes

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java [hunk 1]
```diff
@@ -674,13 +678,19 @@ private int grantOrRevokeRole(GrantRevokeRoleDDL grantOrRevokeRoleDDL)
     return 0;
   }
 
+  private HiveAuthorizationTranslator getAuthorizationTranslator(HiveAuthorizer authorizer) {
+    if(authorizer.getHiveAuthorizationTranslator() == null) {
+      return defaultAuthorizationTranslator ;
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java]: extra space

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAccessController.java [hunk 2]
```diff
@@ -69,4 +69,5 @@ void revokeRole(List<HivePrincipal> hivePrincipals, List<String> roles, boolean
       HiveAccessControlException;
 
   void applyAuthorizationConfigPolicy(HiveConf hiveConf) throws HiveAuthzPluginException;
+
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAccessController.java]: Not needed.

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizerImpl.java [hunk 3]
```diff
@@ -141,19 +137,11 @@ public void applyAuthorizationConfigPolicy(HiveConf hiveConf) throws HiveAuthzPl
   }
 
   @Override
-  public List<HivePrincipal> getHivePrincipals(
-      List<PrincipalDesc> principals) throws HiveException {
-    return AuthorizationUtils.getHivePrincipals(principals);
+  public HiveAuthorizationTranslator getHiveAuthorizationTranslator() throws HiveException {
+    // no customization of this API is done for most Authorization
+    // implementations.
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizerImpl.java]: hint: move to the JavaDoc

**Comment by winningsix** [comment]
Thanks @thejasmn for working on it. Just some minor issues. @sundapeng Can you take a look at this PR? Thank you!

File: ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java (hunk 1 above)
Comment (thejasmn) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java]: fixing

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAccessController.java (hunk 2 above)
Comment (thejasmn) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAccessController.java]: fixing

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizerImpl.java (hunk 3 above)
Comment (thejasmn) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizerImpl.java]: fixing

**Comment by thejasmn** [comment]
addressing the comments

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizationTranslator.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * This is an interface that is not meant for general use, it is targeted to some
+ * specific use cases of Apache Sentry (incubating).
+ * The API uses several classes that are considered internal to Hive, and it is
+ * subject to change across releases.
+ */
+@LimitedPrivate(value = { "Apache Sentry (incubating)" })
+@Evolving
+public interface HiveAuthorizationTranslator {
+
+  public List<HivePrincipal> getHivePrincipals(List<PrincipalDesc> principals)
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizationTranslator.java]: hint: Unify the API as discussed in Jira. Change it to
```
HivePrincipal getHivePrincipal(PrincipalDesc principals)
```

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizationTranslator.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      throws HiveException;
+
+  public List<HivePrivilege> getHivePrivileges(List<PrivilegeDesc> privileges);
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/HiveAuthorizationTranslator.java]: hint: The same as above. Change it to
```
HivePrivilege getHivePrivileges(PrivilegeDesc privilege)
```

**Comment by thejasmn** [comment]
Updated patch addresses additional comments

File: ql/src/java/org/apache/hadoop/hive/ql/security/authorization/DefaultHiveAuthorizationTranslator.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+
+/**
+ * Default implementation of HiveAuthorizationTranslator
+ */
+public class DefaultHiveAuthorizationTranslator implements HiveAuthorizationTranslator {
+
+  @Override
+  public HivePrincipal getHivePrincipal(PrincipalDesc principal) throws HiveException {
+    return AuthorizationUtils.getHivePrincipal(principal);
```
Comment (winningsix) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/DefaultHiveAuthorizationTranslator.java]: It looks wired. Do we still need ```getHivePrincipal``` for the ```AuthorizationUtils```? We can move the implementation of these three APIs to the translator.

Comment (thejasmn) [inline ql/src/java/org/apache/hadoop/hive/ql/security/authorization/DefaultHiveAuthorizationTranslator.java]: Agree, those functions are not being used from anywhere else.

**Pull request closed without merging**